src/ifcopenshell-python/ifcopenshell/validate.py

```diff
@@ -563,9 +564,23 @@ def validate_guid(guid: str) -> Union[str, None]:
         return "Couldn't decompress guid, it's not base64 encoded."
     return None
 
+def to_string_header_entity(header_entity):
+    """Recreate IFC header string representation, like FILE_NAME(...)"""
+    
+    # Prefer native .toString() if available (native IfcOpenShell wrapper)
+    if hasattr(header_entity, 'toString'):
+        return header_entity.toString()
+
```

Comment:
```suggestion
```

First branch 2 times.

Also please change to isinstance() forgot to mention that.

Reply:
[Done](https://github.com/IfcOpenShell/IfcOpenShell/pull/6650/commits/6d2885ef2f7a05b7b17102233f77be4ce87a2453)

simple_spf.file doesn't have a distinctive type for header entity (`tuple` comes the closest); this
```
    if isinstance(header_entity, W.HeaderEntity):
        return header_entity.toString()
    elif isinstance(header_entity, tuple):
```
should catch it. 

Also changed it in the type hinting in the function arguments. I'm still not really convinced about doing that, as it's not enforced anywhere. I saw this is possible at runtime with adding a [typeguard.typechecked](https://typeguard.readthedocs.io/en/stable/api.html#typeguard.typechecked) decorator, but that just results in extra overhead.